File: leetcode/groupAnagrams.java
class Solution {
  public List<List<String>> groupAnagrams(String[] strs) {
      HashMap<String, List<String>> map = new HashMap<>();
      
      for(String str : strs) {
          char[] tmpArray = str.toCharArray();
          Arrays.sort(tmpArray);
          String tmp = new String(tmpArray);
          if(map.containsKey(tmp)) map.get(tmp).add(str);
          else {
              List<String> li = new ArrayList<>();
              li.add(str);
              map.put(tmp, li);
          }
      }
      
      List<List<String>> ans = new ArrayList<>();
      
      for(List<String> li : map.values()) ans.add(li);
      
      return ans;
  }
}

File: leetcode/groupAnagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
 var groupAnagrams = function(strs) {
  let map = new Map()
  
  strs.forEach(el => {
      let arr = el.split('')
      arr.sort()
      let tmp = arr.join('')
      if(map.has(tmp)) map.get(tmp).push(el)
      else map.set(tmp, [el])
  })
  
  let ans = []
  
  map.forEach((val, key) => ans.push(val))
  
  return ans
};